styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -263,6 +266,11 @@ private boolean shouldExecute(RunState runState) {
   }
 
   private void sendDequeue(InstanceState instanceState) {
+    if (!dequeueRateLimiter.tryAcquire()) {
+      LOG.debug("Dequeue rate limited");
+      return;
```

Comment:
This would cause resource usage to be incremented without the instance being dequeued. This happens on the call side of `sendDequeue`. Either return a boolean from this, or check the limited before calling?